### DISCUSSION TITLE
Chats in Folders Excluded from History in Admin User Panel

### DISCUSSION THREAD

**Discussion by El-Tatane** [open-webui/open-webui#7949] [Issues]
# Description
In the function get_chat_list_by_user_id located in backend/open_webui/apps/webui/models/chats.py, the query explicitly filters folder_id to None.

```python
def get_chat_list_by_user_id(  
    self,  
    user_id: str,  
    include_archived: bool = False,  
    skip: int = 0,  
    limit: int = 50,  
) -> list[ChatModel]:  
    with get_db() as db:  
        query = db.query(Chat).filter_by(user_id=user_id).filter_by(folder_id=None)  
     ...
```

This creates an issue where, in the admin user panel, only chats that are not in a folder are displayed. Chats stored in folders are excluded, which is likely not the intended behavior.

# Steps to Reproduce:
	1.	Log in as an admin user.
	2.	Navigate to the user panel to view historical chats from user.
	3.	Observe that only chats without a folder (folder_id=None) are displayed.

# Expected Behavior:
The admin user panel should display all chats, regardless of whether they are in a folder or not.

# Proposed Solution:
	•	Modify the query in get_chat_list_by_user_id to include chats with any folder_id.
	•	Optionally, introduce a parameter to toggle the inclusion of folder-based filtering.

**Comment by rcourtman** [comment]
Looking at your screenshots, the **Degraded** status is because the Pulse agent (installed on `fsn-prod-n01`) cannot reach the other cluster nodes' Proxmox APIs. The errors show:

```
dial tcp [IP]:8006: connect: network is unreachable
```

This indicates the IPs that Proxmox reports for nodes n02/n03 aren't reachable from n01. Common causes:
- Proxmox is reporting internal/wrong IPs (IPv6 link-local, or IPs on a different VLAN)
- Firewall blocking port 8006 between nodes
- Network routing issue

**To fix:**
1. Check what IPs Proxmox advertises: SSH to n01 and run `pvecm nodes` - verify those IPs are reachable from n01
2. Test connectivity: `curl -k https://[n02-ip]:8006/api2/json/version` from n01
3. If Proxmox is reporting wrong IPs, update `/etc/hosts` or fix the node's corosync address in `/etc/pve/corosync.conf`

**Best practice for monitoring a 3-node cluster:**
One agent installed on one node is correct. The agent uses the Proxmox cluster API which provides data for all nodes. The agent just needs network access to all nodes' port 8006.

---

**Re: Docker Swarm** - Not currently supported. Pulse monitors Docker hosts via the Docker API, but Swarm orchestration metadata isn't collected. This would require additional implementation.

**Re: IP addressing changes** - The IPs Pulse shows come from Proxmox's cluster membership. If they're changing unexpectedly, check your corosync configuration and ensure nodes have stable IPs on the cluster network.
